cli/azd/pkg/project/project.go

```diff
@@ -205,7 +229,25 @@ func SaveConfig(ctx context.Context, config config.Config, projectFilePath strin
 
 // Saves the current instance back to the azure.yaml file
 func Save(ctx context.Context, projectConfig *ProjectConfig, projectFilePath string) error {
```

Comment:
If we know we are going to be making mutations on the object and don't want these to be available in the calling instance we may as well just change the func to pass the `ProjectConfig` by value here.  Then you would only need to worry about copying/cloning the services.